inductiva/projects/project.py

```diff
@@ -0,0 +1,240 @@
+        return None
+
+    @staticmethod
+    def _create_model(name):
+        """Creates a project with the given name on the backend."""
+        try:
+            _logger.debug("Creating remote project %s", name)
+            api = projects_api.ProjectsApi(inductiva.api.get_client())
+            response = api.create_project({"name": name})
+            return response.body
+        except ApiException as ex:
+            _logger.error("Failed to create remote project %s",
+                          name,
+                          exc_info=ex)
+            raise RuntimeError(f"Unable to create project {name}") from ex
+
+    def _update_from_api_response(self, model: ProjectModel) -> "Project":
+        """Updates itself with information fetched from the backend."""
+        self._info = ProjectInfo(**model.get("task_status_overview"))
+        self.created_at = model.get("created_at")
+        self.num_tasks = model.get("num_tasks")
+        self.name = model.get("name")
+        self.id = model.get("id")
+
+        return self
+
+    def open(self):
+        """Opens the project.
+
+        Open the project and make it the active one in the calling
+        thread. An opened project will ensure that calls to the
+        `get_current_project` function will return this project.
+        Consecutive calls to this method are idempotent.
+
+        Raises:
+            RuntimeError if another opened project exists, i.e.,
+            `get_current_project` returns something other than None.
+
+        """
+        _logger.debug("Opening project %s", self.name)
+        current_project = get_current_project()
+
+        if current_project is self:
+            _logger.debug("Project is already opened.")
+            return
+
+        if current_project is not None:
+            raise RuntimeError(
+                "Trying to open a project when another is active.")
+
+        self._token = _CURRENT_PROJECT.set(self)
+
+    def close(self):
+        """Closes the project.
+
+        Calls to the `get_current_project` will return `None` after
+        the project is closed.  Consecutive calls to this method are
+        idempotent.
+
+        """
+        if self._token is None:
+            return
+
+        _CURRENT_PROJECT.reset(self._token)
+        self._token = None
+
+    @property
+    def opened(self) -> bool:
+        """Checks if the project is open."""
+        return self._token is not None
+
+    @property
+    def info(self) -> ProjectInfo:
+        """Returns project info.
+
+        It makes no requests to the backend. Instead it uses
+        previously stored information. Therefore it can be
+        outdated. For a method that requests the most recent data from
+        the backend use: `get_info`.
+
+        """
+        return self._info
+
+    def get_info(self) -> ProjectInfo:
+        """Returns and updates project info.
+
+        Updates itself with info from the backend.
+
+        """
```

Comment:
```suggestion
        """Get project information.

        Get updated information on the project. This method
        executes a call to the backend to retrieve the most recent
        information about this project and stores in internally so that
        it becomes available through the `Project.info` property.
        This method is suitable when up-to-date information is required.
        """
```